tests/test_arrow_dataset.py

```diff
@@ -1933,6 +1933,62 @@ def test_dataset_from_file(in_memory, dataset, arrow_file):
     assert dataset_from_file.cache_files == ([{"filename": filename}] if not in_memory else [])
 
 
+@pytest.mark.parametrize("json_format", [None, "jsonl", "json_list_of_dicts", "json_dict_of_lists"])
+@pytest.mark.parametrize("keep_in_memory", [False, True])
+@pytest.mark.parametrize(
+    "features",
+    [
+        None,
+        {"col_1": "string", "col_2": "int64", "col_3": "float64"},
+        {"col_1": "string", "col_2": "string", "col_3": "string"},
+        {"col_1": "int32", "col_2": "int32", "col_3": "int32"},
+        {"col_1": "float32", "col_2": "float32", "col_3": "float32"},
+    ],
+)
+@pytest.mark.parametrize("split", [None, NamedSplit("train"), "train", "test"])
+@pytest.mark.parametrize("path_type", [str, list])
```

Comment:
All these combinations create 100+ tests. I think it added ~5min of time for the CI to run.
Could we make this faster ?

Reply:
I am trying to find out why they are so slow...

Reply:
Tests per-se are not slow. But I think there is some kind of race condition due to `utils\filelock.py::acquire`:
```python
time.sleep(poll_intervall)
```

Reply:
Not a big performance gain after my last commits... :(

Reply:
I have finally removed tests.

Reply:
That did the job to reduce testing time, thanks !

Maybe we can unit-test the json builder for the supported format in test_packaged_modules.py (as for the csv builder) ? (in another PR)